tests/testdata/input/remove-comments/test_comments.md

```diff
@@ -0,0 +1,11 @@
+Not a comment
+%% Comment should be removed %%
+```md
+%% Comment in code block should be kept %%
+```
+
+%%
+This is
+a block comment
+that should be removed
+%%
```

Comment:
Does your current implementation handle non-text elements within the comment? As in, Obsidian treats this entire block as a comment:

````
%%

None of the below should be visible in output:

## Header

Code block:

```
foo
```

---

%%
````

Reply:
While my current implementation handles most of this edge case it struggles with the code block (removing everything except the opening and closing ```. I will get started on this fix.